packages/material-ui/src/styles/muiStyled.js

```diff
@@ -0,0 +1,62 @@
+import styled from '@material-ui/styled-engine';
+import { propsToClassKey } from '@material-ui/styles';
+import defaultTheme from './defaultTheme';
+
+const getStyleOverrides = (name, theme) => {
+  let styleOverrides = {};
+
+  if (
+    theme &&
+    theme.components &&
+    theme.components[name] &&
+    theme.components[name].styleOverrides
+  ) {
+    styleOverrides = theme.components[name].styleOverrides;
+  }
+
+  return styleOverrides;
+};
+
+const getVariantStyles = (name, theme) => {
+  let variants = [];
+  if (theme && theme.components && theme.components[name] && theme.components[name].variants) {
+    variants = theme.components[name].variants;
+  }
+
+  const variantsStyles = {};
+
+  variants.forEach((definition) => {
+    const key = propsToClassKey(definition.props);
+    variantsStyles[key] = definition.style;
+  });
+
+  return variantsStyles;
+};
+
+const shouldForwardProp = (prop) => prop !== 'state' && prop !== 'theme';
+
+const muiStyled = (el, styledParams, muiConfig) => {
```

Comment:
`element`? We shouldn't use these short forms unless these are proper terms that only have a very long alternative e.g `mui`.